src/dataviews/dataview-model-base.js

```diff
@@ -76,6 +78,7 @@ module.exports = Model.extend({
 
   _initBinds: function () {
     this.listenTo(this.layer, 'change:visible', this._onLayerVisibilityChanged);
+    this.listenTo(this.layer, 'change:source', this._onLayerVisibilityChanged);
```

Comment:
Hmm not sure I follow you… do you mean that the visibility of the layer should influence what `getSourceId` returns? E.g. if layer is not visible return the own source instead?